var/spack/repos/builtin/packages/mesa/package.py

```diff
@@ -111,22 +149,27 @@ def configure_args(self):
         else:
             args.append('--disable-gallium-osmesa')
 
-        if '+glx' in spec:
+        if glx_enabled:
             num_frontends += 1
-            if '+egl' in spec:
+            if dri_enabled:
                 args.append('--enable-glx=dri')
             else:
                 args.append('--enable-glx=gallium-xlib')
             args_platforms.append('x11')
         else:
             args.append('--disable-glx')
 
-        if '+egl' in spec:
+        if dri_enabled:
```

Comment:
This whole block can be simplified now and can ignore any dri related things.  They were mixed in before because egl was synonymous with dri, but now they're orthogonal:
```python
if egl_enabled:
  num_frontends +=1
  args.extend(['--enable-egl', '--enable-gbm'])
  args_platforms.append('surfaceless')
else:
  args.extend(['--disable-egl', '--disable-gbm')
```